packages/db-objects-redis/lib/objects/lua-v3/redlock_acquire.lua

```diff
@@ -0,0 +1,12 @@
+-- REDLOCK SCRIPT
+-- call with key, value, ms (for set px)
+local key = KEYS[1]
+
+if redis.call("exists", key) == 1 then
+  -- lock already acquired by someone could be us too (we can only extend)
+  return 0
+end
+
+redis.call("set", key, KEYS[2], "PX", KEYS[3])
```

Comment:
btw... what happens on error here? set shoukd return 1 on success or??? Can we check that?

Reply:
And: really also for v3?

Reply:
Think it wouldn't hurt if we have a v3 host and two vX hosts which then would run with  protocol v3 and the vX hosts have a new code part which should only be executed by one of all, then it would work correctly.

Reply:
Regarding the error, the command should throw because it is no `pcall`.

Reply:
No scripts will always be registered, so that should work, but yes ok makes sense

Reply:
Yes registered but executing will result in a log entry unknown script xy

Reply:
aahhhhh ... I think the script register needs a "--design" in first place OR a "--func" so that ... seee ObjectInMemServerRedis L271ff ... maybe a "--func" could work? O radd a redlock else there

Reply:
I have the redlock else but I mean if the controller which starts the sim does not have the code but newer clients the to execute the redlich script

Reply:
This is exactly what I mean ... the old controller do not knwo --REDLOCK .. they just jknow --func and --design ... 

Infact you could do 

--func: function() {return {redlock:true};}` 

because old controller would not care really about it and you can still compare this as special case in v4?

Reply:
ah yes makes sense

Reply:
hm but func call always expects at least 4 args etc and subscribe on evicted in objects also not present, so probably better say if we have a host < 4 (actually useSets false) do not make the priamry magic

Reply:
in fcat this is not the same because these sc ripts here would never be used in a "object view scenario", but yes better to have a clean separation ... v4 can have it, v3 not ... done